src/controllers/products/buyerViewSingleProductController.js

```diff
@@ -23,6 +25,39 @@ export const viewSingleProduct = asyncWrapper(async (req, res) => {
   }
   return res.status(200).json({
     status: req.t('success'),
-    data: product
+    data: product,
   });
 });
+
+export const recommendedProducts = asyncWrapper(async (req, res) => {
+  const { id } = req.user;
+  const wishlist = await db.wishlist.findOne({ where: { userId: id }, attributes: ['productId'] });
+  if (!wishlist) {
+    return res.status(404).json({
+      status: req.t('fail'),
+      message: req.t('no_wishlist'),
+    });
+  }
+  const { productId } = wishlist.dataValues;
+  const reco = await productId.map(async (id) => {
```

Comment:
Work on your naming conventions (Use names that are easily understandable and use camelCase) like 
1. `reco`, I think you wanted to refer to recommendation
2. whishlist : `wishList` and so on...

Reply:
We can arrange a call tomorrow morning to share ideas on this algorithm of product recommendations

Reply:
sure we can do that but I have changed variables and now I am using readable variable